README.md

```diff
@@ -304,20 +309,44 @@ request.get();
 // This will throw if the request takes longer than 15 seconds.
 ```
 
-##### Credentials (browser only)
-HTTP requests made from a browser have an added restriction - secure cookies are
-not sent by default on cross-origin requests. To include these secure cookies
-when sending a request, set `withCredentials` to `true`. Although this only
-applies to browsers, it's included in the platform-independent API because it
-has no effect on the other platforms.
+#### Request & Response Interception
+All of the request classes have hooks that allow request and response
+interception.
+
+> Although hooks for interception are available on the request classes, their
+> purpose is to enable an API on the [`Client` class](#http-client) for
+> intercepting every request created by the client, which is much more useful.
+
+**Request interception** occurs right before the request is dispatched, at which
+point changes (async if necessary) to the request instance can be made.
 
 ```dart
-Request request = new Request()
-  ..uri = Uri.parse('https://otherhost.com/notes/')
-  ..withCredentials = true;
-await request.get();
+Request request = new Request();
+
+// Register a hook to intercept the request.
+request.requestInterceptor = (Request request) async {
+  // Modify the request as necessary.
+};
+```
+
+**Response interception** occurs after the response is received but before it is
+delivered to the caller. At this point, a finalized version of the request can
+be inspected and the response instance can be modified, augmented, or replaced.
+Additionally, a `RequestException` instance will be available if one occurred.
+Again, this interception can be async if necessary.
+
+```dart
+Request request = new Request();
+
+// Register a hook to intercept the response.
+request.responseInterceptor =
+    (FinalizedRequest request, BaseResponse response,
+    [RequestException exception]) async {
+  // Return a `BaseResponse` instance, modified as necessary.
```

Comment:
i'd vote for both interceptors to either modify the incoming request / response in place or explicitly return an object instance instead of request and response pattern being different